tests/python/unittest/common.py

```diff
@@ -272,6 +272,27 @@ def teardown():
     mx.nd.waitall()
 
 
+def with_post_test_cleanup():
+    """
+    Helper function that cleans up memory by releasing it from memory pool
+    Required especially by large tensor tests that have memory footprints in GBs.
+    """
+    def test_helper(orig_test):
+        @make_decorator(orig_test)
+        def test_new(*args, **kwargs):
+            logger = default_logger()
+            try:
+                orig_test(*args, **kwargs)
+            except:
+                logger.info(test_msg)
+                raise
+            finally:
+                mx.nd.waitall()
+                mx.cpu().empty_cache()
```

Comment:
are the tests only for cpu context ?

Reply:
Ya. We don't test for gpu (as large tensor is not supported)